src/modelframe.jl

```diff
@@ -68,9 +68,15 @@ end
 
 missing_omit(data::T, formula::AbstractTerm) where T<:ColumnTable =
     missing_omit(NamedTuple{tuple(termvars(formula)...)}(data))
-
+    
 function ModelFrame(f::FormulaTerm, data::ColumnTable;
                     model::Type{M}=StatisticalModel, contrasts=Dict{Symbol,Any}()) where M
+    
+    msg = checknamesexist( f, data )
+    if msg != ""
+        throw(ArgumentError(msg))
+    end
```

Comment:
This is now redundant with the `concrete_term` checks (since those get triggered by the `schema` call a few lines below).  Or, actually, maybe not, since `missing_omit` may be trying to access the missing columns.  Either way I don't think it really does any harm by being here, too, so you get all the missing variables at once.